cli/send.go

```diff
@@ -244,3 +255,107 @@ var SendCmd = &cli.Command{
 		return nil
 	},
 }
+
+func handleCSVSend(cctx *cli.Context, csvFile string) error {
+	srv, err := GetFullNodeServices(cctx)
+	if err != nil {
+		return err
+	}
+	defer srv.Close() //nolint:errcheck
+
+	ctx := ReqContext(cctx)
+
+	// Add logic to handle --from address
+	var fromAddr address.Address
+	if from := cctx.String("from"); from != "" {
+		addr, err := address.NewFromString(from)
+		if err != nil {
+			return err
+		}
+		fromAddr = addr
+	} else {
+		defaddr, err := srv.FullNodeAPI().WalletDefaultAddress(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get default address: %w", err)
+		}
+		fromAddr = defaddr
+	}
+
+	// Print sending address
+	_, _ = fmt.Fprintf(cctx.App.Writer, "Sending messages from: %s\n", fromAddr.String())
+
+	fileReader, err := os.Open(csvFile)
+	if err != nil {
+		return xerrors.Errorf("read csv: %w", err)
+	}
+	defer func() {
+		if err := fileReader.Close(); err != nil {
+			log.Errorf("failed to close csv file: %v", err)
+		}
+	}()
+
+	r := csv.NewReader(fileReader)
+	records, err := r.ReadAll()
+	if err != nil {
+		return xerrors.Errorf("read csv: %w", err)
+	}
+
+	// Validate header
+	if len(records) == 0 ||
+		strings.TrimSpace(records[0][0]) != "Recipient" ||
+		strings.TrimSpace(records[0][1]) != "FIL" ||
+		strings.TrimSpace(records[0][2]) != "Method" ||
+		strings.TrimSpace(records[0][3]) != "Params" {
+		return xerrors.Errorf("expected header row to be \"Recipient,FIL,Method,Params\"")
+	}
+
+	for i, e := range records[1:] {
+		var params SendParams
```

Comment:
given all the validation that happens in here, I think you should consider accumulating a `[]SendParams` as a map of the original records, so your first pass is to validate and build the params, then the second pass is to perform the send on each of them so there's fewer opportunities for errors in the middle of sending